cpp/src/slice2java/Gen.cpp

```diff
@@ -4238,10 +4240,9 @@ Slice::Gen::ProxyVisitor::visitInterfaceDefEnd(const InterfaceDefPtr& p)
         "Raises a local exception if a communication error occurs.\n"
         "@param obj The untyped proxy.\n"
         "@return A proxy for this type, or null if the object does not support this type.");
-    out << nl << "static " << p->name() << "Prx checkedCast(com.zeroc.Ice.ObjectPrx obj)";
+    out << nl << "static " << prxName << " checkedCast(com.zeroc.Ice.ObjectPrx obj)";
```

Comment:
For mapped proxies, we used to delegate to generic functions on `ObjectPrx` like `_checkedCast`.
These were publicly visible, and used ugly reflection stuff.

Now, we generate all the code we need in the mapped proxy class itself.
This lets us use a constructor to create the proxy, instead of needing reflection.

It's more performant (without reflection), keeps `ObjectPrx` cleaner, and IMO easier to follow.